django/db/models/base.py

```diff
@@ -741,12 +741,16 @@ def refresh_from_db(self, using=None, fields=None, from_queryset=None):
 
         # Clear cached relations.
         for field in self._meta.related_objects:
-            if field.is_cached(self):
+            if (fields is None or field.name in fields) and field.is_cached(self):
```

Comment:
`fields` is a list, using `field.name in fields` is O(n). I know n will normally be small, but I wonder if we could speed it up a little. Does it work if we instead turn `fields` into a set on line 695, giving an O(1) runtime for lookups?

Reply:
Good idea, making it a `set()` seems to work just fine so I'm going for it.